ddl/ddl.go

```diff
@@ -468,54 +472,91 @@ func (d *ddl) newDeleteRangeManager(mock bool) delRangeManager {
 
 // Start implements DDL.Start interface.
 func (d *ddl) Start(ctxPool *pools.ResourcePool) error {
-	logutil.BgLogger().Info("[ddl] start DDL", zap.String("ID", d.uuid), zap.Bool("runWorker", RunWorker))
+	var err error
+	logutil.BgLogger().Info("[ddl] start DDL", zap.String("ID", d.uuid), zap.Bool("runWorker", config.GetGlobalConfig().Instance.TiDBEnableDDL.Load()))
 
 	d.wg.Run(d.limitDDLJobs)
 	d.sessPool = newSessionPool(ctxPool, d.store)
+	d.workers = make(map[workerType]*worker, 2)
```

Comment:
Then you will still enable the runner. I suggest you to make `*ddl` having a real `Start` and `Stop`. `Start() at L470 is more like an one-shot `Run()/Begin()`.